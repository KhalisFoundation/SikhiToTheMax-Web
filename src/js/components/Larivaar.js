import React from 'react';
import PropTypes from 'prop-types';
import { LARIVAAR_ASSIST_COLOR } from '../constants';

export default class Larivaar extends React.PureComponent {
  static defaultProps = {
    larivaarAssist: false,
    enable: true,
  };

  static propTypes = {
    larivaarAssist: PropTypes.bool,
    enable: PropTypes.bool,
    children: PropTypes.string.isRequired,
  };

  render() {
    const { larivaarAssist, enable, children } = this.props;
    const larivaarAssistColor = larivaarAssist ? LARIVAAR_ASSIST_COLOR : '';
    return (
      <React.Fragment>
        {enable
          ? children.split(' ').map(
              (val, indexVal) => {
                if(val.indexOf('॥') !== -1 || val.indexOf(']') !== -1) {
                  return `${val} `
                } else {
                  var breakSupportedCharList = [];

                  // before rendering each word
                  // Search for matras, lagakhar, dut akhar etc in it to preserve word break
                  // we can't use css word-break - break all property anymore
                  // In that case we have to put wbr with every character
                  // So we are checking every character
                  var splitArr = val.split('')
                  for(let i=0;i<splitArr.length;i++) {
                    // checking for (Right side matras) Lavaa, Dulavaa, Aunkar, Dulainkar, Bihari, Kanna, Bindi and Tippi
                    // Will optimize this condition
                    // Remaining check for - Haf Characters(Dutt akhar) and combination of Right Side Matras and Half characters
                    // Many complexwords are present which containscombination of Right SIde matras with Half characters and Bindi, Tippi, Adhak
                    // need to test thoroughly for this condition
                    // Remainig check for Kanaa + Bindi Combination
                    // Check for unicode would be diffrent as there is no left side matra in Unicode - Unicode have right side matras only
                    // and combination of Matras + Bindi + Tippi + Half character is diffrent than fnts we use
                    // 
                    // Pairri rarra is represented by "®" in gurbani font 

                    // Facepalm
                    // 1. other form of Tippi - µ
                    // 2. issue: there are two types of Pairri Rarra - One "R" and other "®"
                    // 
                    // thats why I hate punjabi fonts - No consistency
                    // 
                    // Done:
                    // Kanna Bindi - single entity
                    // Many complex words are present which contains combination of Right Side matras with Half characters and Bindi, Tippi, Adhak
                    // and combination of Matras + Bindi + Tippi + Half character is diffrent than fonts we use
                    // Sihari + char + Half char + Bindi/Tippi
                    if(splitArr[i+1] && ((splitArr[i+1].indexOf("Y")  != -1) || (splitArr[i+1].indexOf("y")  != -1) || (splitArr[i+1].indexOf("w")  != -1) || (splitArr[i+1].indexOf("W")  != -1) || (splitArr[i+1].indexOf("I")  != -1)  || (splitArr[i+1].indexOf("u")  != -1) || (splitArr[i+1].indexOf("U")  != -1) || (splitArr[i+1].indexOf("N")  != -1) || (splitArr[i+1].indexOf("M")  != -1) || (splitArr[i+1].indexOf("´")  != -1) || (splitArr[i+1].indexOf("R")  != -1) || (splitArr[i+1].indexOf("Í")  != -1) || (splitArr[i+1].indexOf("µ")  != -1) || (splitArr[i+1].indexOf("@")  != -1))) {
                      // Dut akhar(Pairi Rarra) + Bindi or Tippi as one entity
                      if(splitArr[i+1].indexOf("´")  != -1 && splitArr[i+2] && ((splitArr[i+2].indexOf("M")  != -1) || (splitArr[i+2].indexOf("N")  != -1))) {
                        breakSupportedCharList.push(
                          <span key={i} className = "wordBreakSpecialChar">
                            {splitArr[i] + splitArr[i+1] + splitArr[i+2]}<wbr />
                          </span>
                        )
                        i=i+2;
                      } else if(splitArr[i+1].indexOf("R")  != -1 && splitArr[i+2] && ((splitArr[i+2].indexOf("w")  != -1) || (splitArr[i+1].indexOf("W")  != -1))) {
                        // pairri Rarra + Kanna
                        breakSupportedCharList.push(
                          <span key={i} className = "wordBreakSpecialChar">
                            {splitArr[i] + splitArr[i+1] + splitArr[i+2]}<wbr />
                          </span>
                        )
                        i=i+2;
                      } else if(splitArr[i+1].indexOf("´")  != -1 && splitArr[i+2] && ((splitArr[i+2].indexOf("Y")  != -1) || (splitArr[i+2].indexOf("y")  != -1) || (splitArr[i+2].indexOf("w")  != -1) || (splitArr[i+2].indexOf("I")  != -1)  || (splitArr[i+2].indexOf("u")  != -1) || (splitArr[i+2].indexOf("U")  != -1) || (splitArr[i+2].indexOf("N")  != -1) || (splitArr[i+2].indexOf("M")  != -1) || (splitArr[i+1].indexOf("µ")  != -1))) {
                        breakSupportedCharList.push(
                          <span key={i} className = "wordBreakSpecialChar">
                            {splitArr[i] + splitArr[i+1] + splitArr[i+2]}<wbr />
                          </span>
                        )
                        i=i+2;
                      } else if(splitArr[i+1].indexOf("@")  != -1 && splitArr[i+2] && ((splitArr[i+2].indexOf("Y")  != -1) || (splitArr[i+2].indexOf("y")  != -1) || (splitArr[i+2].indexOf("w")  != -1) || (splitArr[i+2].indexOf("I")  != -1)  || (splitArr[i+2].indexOf("u")  != -1) || (splitArr[i+2].indexOf("U")  != -1) || (splitArr[i+2].indexOf("N")  != -1) || (splitArr[i+2].indexOf("M")  != -1) || (splitArr[i+1].indexOf("µ")  != -1))) {
                        breakSupportedCharList.push(
                          <span key={i} className = "wordBreakSpecialChar">
                            {splitArr[i] + splitArr[i+1] + splitArr[i+2]}<wbr />
                          </span>
                        )
                        i=i+2;
                      } else if((splitArr[i+1].indexOf("´")  != -1) && splitArr[i+2] && ((splitArr[i+1].indexOf("M")  != -1) || (splitArr[i+1].indexOf("N")  != -1) || (splitArr[i+1].indexOf("µ")  != -1))) {
                        // right side matra + Bindi + Tippi
                        breakSupportedCharList.push(
                          <span key={i} className = "wordBreakSpecialChar">
                            {splitArr[i] + splitArr[i+1] + splitArr[i+2]}<wbr />
                          </span>
                        )
                        i=i+2;
                      } else if((splitArr[i+1].indexOf("M")  != -1) || (splitArr[i+1].indexOf("N")  != -1)  || (splitArr[i+1].indexOf("µ")  != -1)) {
                        // right side matra + Bindi + Tippi
                        breakSupportedCharList.push(
                          <span key={i}>
                            {splitArr[i] + splitArr[i+1]}<wbr />
                          </span>
                        )
                        i=i+1;
                      } else if(splitArr[i+1].indexOf("´")  != -1) {
                        breakSupportedCharList.push(
                          <span key={i} className = "wordBreakSpecialChar">
                            {splitArr[i] + splitArr[i+1]}<wbr />
                          </span>
                        )
                        i++;
                      } else {
                        breakSupportedCharList.push(
                          <span key={i}>
                            {splitArr[i] + splitArr[i+1]}<wbr />
                          </span>
                        )
                        i++;  
                      }

                    } else if(splitArr[i].indexOf("i") != -1 ) {
                      // handle bindi, tippi, dutt akhar with sihari 3 characters together
                      // pending:
                      // handle bindi, tippi, dutt akhar with sihari 4 character together
                      if(splitArr[i+2] && splitArr[i+3] && (splitArr[i+2].indexOf("´")  != -1) && ((splitArr[i+3].indexOf("N")  != -1) || splitArr[i+3].indexOf("M")  != -1)) {
                        // checking for sihari matra with pairri Yayya and Bindi or adhak 
                        breakSupportedCharList.push(
                          <span key={i} className = "wordBreakSpecialChar">
                            {splitArr[i] + splitArr[i+1] + splitArr[i+2] + splitArr[i+3]}<wbr />
                          </span>
                        )
                        i=i+3;
                      } else if(splitArr[i+2] && splitArr[i+3] && (splitArr[i+2].indexOf("´")  != -1)) {
                        // checking for sihari matra with pairri Yayya
                        breakSupportedCharList.push(
                          <span key={i} className = "wordBreakSpecialChar">
                            {splitArr[i] + splitArr[i+1] + splitArr[i+2]}<wbr />
                          </span>
                        )
                        i=i+2;
                      } else if(splitArr[i+2] && splitArr[i+3] && (splitArr[i+2].indexOf("®")  != -1) && ((splitArr[i+3].indexOf("N")  != -1) || splitArr[i+3].indexOf("M")  != -1)) {
                        // checking for sihari matra and pairri rarra other form - bindi, tippi 
                        breakSupportedCharList.push(
                          <span key={i}>
                            {splitArr[i] + splitArr[i+1] + splitArr[i+2] + splitArr[i+3]}<wbr />
                          </span>
                        )
                        i=i+3;
                      } else if(splitArr[i+2] && (splitArr[i+2].indexOf("®")  != -1)) {
                        // checking for sihari matra and pairri rarra only
                        breakSupportedCharList.push(
                          <span key={i}>
                            {splitArr[i] + splitArr[i+1] + splitArr[i+2]}<wbr />
                          </span>
                        )
                        i=i+2;
                      } else if((splitArr[i+1].indexOf("M")  != -1) || (splitArr[i+1].indexOf("N")  != -1)  || (splitArr[i+1].indexOf("µ")  != -1)) {
                        // right side matra + Bindi + Tippi
                        breakSupportedCharList.push(
                          <span key={i}>
                            {splitArr[i] + splitArr[i+1]}<wbr />
                          </span>
                        )
                        i=i+1;
                      } else {
                        // checking for sihari matra 
                        breakSupportedCharList.push(
                          <span key={i}>
                            {splitArr[i] + splitArr[i+1]}<wbr />
                          </span>
                        )
                        i++;
                      }
                    } else {
                      // check for all muktas
                      breakSupportedCharList.push(
                        <span key={i}>
                          {splitArr[i]}<wbr />
                        </span>
                      )  
                    }
                  }

                  if(breakSupportedCharList.length) {
                    return <span key={indexVal} style={{ color: indexVal % 2 === 1 ? larivaarAssistColor : '' }}>{breakSupportedCharList.map((element) => {
                      return element
                    })}</span>
                  } else {
                    return (
                      <span key={indexVal}>
                        {val}
                      </span>
                    )  
                  }
                }
              }
            )
          : children}
      </React.Fragment>
    );
  }
}
