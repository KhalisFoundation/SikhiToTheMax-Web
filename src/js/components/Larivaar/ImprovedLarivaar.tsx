import React, { memo } from 'react';

import LarivaarWord from './Word';
import HighlightedSearchResult from '../SearchResults/HighlightedResult';

export interface ILarivaarProps {
  larivaarAssist?: boolean;
  startIndex?: number;
  endIndex?: number;
  enable?: boolean;
  unicode: boolean;
  children: string;
}

function Larivaar(props: ILarivaarProps) {
  const {
    startIndex,
    endIndex,
    larivaarAssist = false,
    enable = true,
    children,
    unicode,
  } = props;

  if (!enable) {
    return (
      <HighlightedSearchResult startIndex={startIndex} endIndex={endIndex}>
        {children}
      </HighlightedSearchResult>
    );
  }

  return (
    <>
      {children.split(' ').map((word, index) => {
        if (['рее', ']'].some(v => word.includes(v))) {
          return `${word} `;
        }

        return (
          <LarivaarWord
            startIndex={startIndex}
            endIndex={endIndex}
            key={index}
            word={word}
            unicode={unicode}
            larivaarAssist={larivaarAssist}
            index={index}
          />
        );
      })}
    </>
  );
}

export default memo(Larivaar);
