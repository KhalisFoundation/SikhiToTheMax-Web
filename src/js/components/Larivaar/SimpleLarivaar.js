import React from 'react';
import PropTypes from 'prop-types';
import { LARIVAAR_ASSIST_COLOR } from '../../constants';

export default class Larivaar extends React.PureComponent {
  static defaultProps = {
    larivaarAssist: false,
    enable: true,
  };

  static propTypes = {
    larivaarAssist: PropTypes.bool,
    enable: PropTypes.bool,
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node])
      .isRequired,
  };

  render() {
    const { larivaarAssist, enable, children } = this.props;

    const larivaarAssistColor = larivaarAssist ? LARIVAAR_ASSIST_COLOR : '';
    return enable === false
      ? children
      : children.split(' ').map(
          (word, index) =>
            ['рее', ']'].some(v => word.includes(v)) ? (
              `${word} `
            ) : (
              <span
                key={index}
                style={{
                  color: index % 2 === 1 ? larivaarAssistColor : '',
                }}
              >
                {word}
              </span>
            )
        );
  }
}
