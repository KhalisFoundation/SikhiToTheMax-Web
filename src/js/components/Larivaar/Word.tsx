import React, { memo } from 'react';

import {
  fixLarivaarUnicode,
  fixLarivaarGurmukhiFont
} from './util';

export interface ILarivaarWordProps {
  word: string;
  unicode: boolean;
  larivaarAssist?: boolean;
  larivaarAssistColor: string;
  index: number;
  highlightIndex?: Array<number>;
  highlight?: boolean;
  visraamClass: string;
}

const LarivaarWord: React.FC<ILarivaarWordProps> = (props) => {
  const {
    highlightIndex,
    word,
    unicode,
    larivaarAssist,
    larivaarAssistColor,
    index,
    highlight,
    visraamClass,
  } = props;

  const isOddIdx = index % 2 === 1;
  const isColoredLarivaarAssist = larivaarAssist && isOddIdx;
  const segments = unicode
    ? fixLarivaarUnicode(word)
    : fixLarivaarGurmukhiFont(word);

  const assignAkharColor = (node: HTMLElement) => {
    if (node) {
      if (isColoredLarivaarAssist) {
        node.style.setProperty('color', larivaarAssistColor, 'important');
      } else {
        node.style.setProperty('color', '');
      }
    }
  }

  return (
    <span className={visraamClass}>
      {segments.map((item, i) => {
        const key = `${index}.${i}`;
        let akharClass = '';

        if (isOddIdx) {
          akharClass += 'larivaar-word';
          if (isColoredLarivaarAssist) {
            akharClass += ' larivaar-assist-word';
          }
        }

        // If this is a search result
        if (highlightIndex !== undefined) {
          if (highlight || (highlightIndex.includes(index))) {
            akharClass += ' search-highlight-word';
          }
        }

        // handle space break for this special character
        if (item.includes('Â´')) {
          // currently react don't support assigning important as inline
          // so need to use this hack of reference
          return (
            <span
              ref={assignAkharColor}
              key={key}
              className={akharClass}
              style={{ display: 'inline-block' }}
            >
              {item}
              <wbr />
            </span>
          );
        }

        // currently react don't support assigning important as inline
        // so need to use this hack of reference
        return (
          <span
            key={key}
            className={akharClass}>
            <span ref={assignAkharColor}>
              {item}
              <wbr />
            </span>
          </span>
        );
      })}
    </span>
  );
}

export default memo(LarivaarWord);
