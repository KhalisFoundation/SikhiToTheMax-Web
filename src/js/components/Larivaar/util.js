const matrasThatAppearAtRightSideOfChar = 'wIuUyYoOW';
const halfCharThatAppearAtRightSideOfChar = 'Í´R@˜®';
const nasalSoundCaharacter = 'NMµ';
const sihari = 'i';
const allMatrasHalfCharAndNasalSoundChar =
  sihari +
  matrasThatAppearAtRightSideOfChar +
  halfCharThatAppearAtRightSideOfChar +
  nasalSoundCaharacter;
const unicodeMatras = 'ਾਿੀੁੂੇੈੋੌੰਂ';

// Look for consonants and break word
export function fixLarivaarGurmukhiFont(str) {
  let arrWordBreak = [];
  // search and break till next consonant
  let segmentedStr = '';

  for (let i = 0; i < str.length; i++) {
    segmentedStr += str[i];
    // handle sihari exeption
    // add to the segment string and ignore other checks
    if (isSihari(str[i])) {
      continue;
    }
    if (isConsonant(str[i + 1]) || isSihari(str[i + 1])) {
      arrWordBreak.push(segmentedStr);
      segmentedStr = '';
    }
  }
  return arrWordBreak;
}

// In unicode All matras are right hand sided
// and matras comes after main consonant and half character
// we need to break word after Matras not consonant
export function fixLarivaarUnicode(str) {
  let arrWordBreak = [];
  // search and break till next consonant
  let segmentedStr = '';

  for (var i = 0; i < str.length; i++) {
    segmentedStr += str[i];
    // exception for half character
    if (str[i + 1] == '੍') {
      segmentedStr += str[i + 1];
      i = i + 1;
      continue;
    }
    if (!isUnicodeConsonant(str[i + 1])) {
      arrWordBreak.push(segmentedStr);
      segmentedStr = '';
    }
  }
  return arrWordBreak;
}

const isConsonant = char =>
  allMatrasHalfCharAndNasalSoundChar.includes(char) === false;
const isUnicodeConsonant = char => unicodeMatras.includes(char) === false;
const isSihari = char => char === sihari;
