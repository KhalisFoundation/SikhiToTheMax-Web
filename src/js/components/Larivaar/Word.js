import React from 'react';
import PropTypes from 'prop-types';
import { fixLarivaarUnicode, fixLarivaarGurmukhiFont } from './util';

/**
 *
 *
 * @export
 * @class LarivaarWord
 * @augments {React.PureComponent<LarivaarWordProps>}
 */
export default class LarivaarWord extends React.PureComponent {
  /**
   * @typedef {object} LarivaarWordProps
   * @property {string} word
   * @property {boolean} [unicode]
   * @property {string} larivaarAssistColor
   * @property {number} index
   */

  static propTypes = {
    word: PropTypes.string.isRequired,
    unicode: PropTypes.bool,
    larivaarAssistColor: PropTypes.string.isRequired,
    index: PropTypes.number.isRequired,
  };

  render() {
    const { word, unicode, larivaarAssistColor, index } = this.props;

    let segments = unicode
      ? fixLarivaarUnicode(word)
      : fixLarivaarGurmukhiFont(word);

    return segments.map((item, i) => {
      const color = index % 2 === 1 ? larivaarAssistColor : '';
      const key = `${index}.${i}`;

      if (item.includes('Â´')) {
        // handle space break for this special character
        return (
          <span
            key={key}
            style={{
              color,
              display: 'inline-block',
            }}
          >
            {item}
            <wbr />
          </span>
        );
      } else {
        return (
          <span key={key} style={{ color }}>
            <span>
              {item}
              <wbr />
            </span>
          </span>
        );
      }
    });
  }
}
