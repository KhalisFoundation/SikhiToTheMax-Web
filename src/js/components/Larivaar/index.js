import React from 'react';
import PropTypes from 'prop-types';
import { LARIVAAR_ASSIST_COLOR } from '../../constants';
import { fixLarivaar } from './util';

export default class Larivaar extends React.PureComponent {
  static defaultProps = {
    larivaarAssist: false,
    enable: true,
  };

  static propTypes = {
    larivaarAssist: PropTypes.bool,
    enable: PropTypes.bool,
    unicode: PropTypes.bool,
    children: PropTypes.string.isRequired,
  };

  render() {
    const { larivaarAssist, enable, children, unicode } = this.props;
    const larivaarAssistColor = larivaarAssist ? LARIVAAR_ASSIST_COLOR : '';
    return (
      <React.Fragment>
        {enable
          ? children.split(' ').map((val, indexVal) => {
              if (val.indexOf('рее') !== -1 || val.indexOf(']') !== -1) {
                return `${val} `;
              } else {
                return fixLarivaar(val, unicode, larivaarAssistColor, indexVal);
              }
            })
          : children}
      </React.Fragment>
    );
  }
}
