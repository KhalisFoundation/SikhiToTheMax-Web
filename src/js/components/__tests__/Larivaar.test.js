import React from 'react';
import { shallow } from 'enzyme';
import Larivaar from '../Larivaar';

describe('<Larivaar />', () => {
    it('renders children when disabled', () => {
        const wrapper = shallow(
            <Larivaar enable={false}>Vahi Guru</Larivaar>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders concatenated children when enabled', () => {
        const wrapper = shallow(
            <Larivaar>Vahi Guru</Larivaar>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders odd words as orange when larivaarAssist is enabled', () => {
        const wrapper = shallow(
            <Larivaar larivaarAssist={true}>Vahi Guru</Larivaar>
        );

        expect(wrapper).toMatchSnapshot();
    })
});