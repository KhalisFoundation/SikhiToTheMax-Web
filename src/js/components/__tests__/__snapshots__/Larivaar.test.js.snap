// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Larivaar /> renders children when disabled 1`] = `
<React.Fragment>
  Vahi Guru
</React.Fragment>
`;

exports[`<Larivaar /> renders concatenated children when enabled 1`] = `
<React.Fragment>
  <span
    key="0.0"
    style={
      Object {
        "color": "",
      }
    }
  >
    <span
      key="0"
    >
      V
      <wbr />
    </span>
  </span>
  <span
    key="0.1"
    style={
      Object {
        "color": "",
      }
    }
  >
    <span
      key="1"
    >
      a
      <wbr />
    </span>
  </span>
  <span
    key="0.2"
    style={
      Object {
        "color": "",
      }
    }
  >
    <span
      key="2"
    >
      h
      <wbr />
    </span>
  </span>
  <span
    key="1.0"
    style={
      Object {
        "color": "",
      }
    }
  >
    <span
      key="0"
    >
      Gu
      <wbr />
    </span>
  </span>
  <span
    key="1.1"
    style={
      Object {
        "color": "",
      }
    }
  >
    <span
      key="1"
    >
      ru
      <wbr />
    </span>
  </span>
</React.Fragment>
`;

exports[`<Larivaar /> renders odd words as orange when larivaarAssist is enabled 1`] = `
<React.Fragment>
  <span
    key="0.0"
    style={
      Object {
        "color": "",
      }
    }
  >
    <span
      key="0"
    >
      V
      <wbr />
    </span>
  </span>
  <span
    key="0.1"
    style={
      Object {
        "color": "",
      }
    }
  >
    <span
      key="1"
    >
      a
      <wbr />
    </span>
  </span>
  <span
    key="0.2"
    style={
      Object {
        "color": "",
      }
    }
  >
    <span
      key="2"
    >
      h
      <wbr />
    </span>
  </span>
  <span
    key="1.0"
    style={
      Object {
        "color": "#f39c1d",
      }
    }
  >
    <span
      key="0"
    >
      Gu
      <wbr />
    </span>
  </span>
  <span
    key="1.1"
    style={
      Object {
        "color": "#f39c1d",
      }
    }
  >
    <span
      key="1"
    >
      ru
      <wbr />
    </span>
  </span>
</React.Fragment>
`;
